src/tss2-fapi/api/Fapi_CreateNv.c

```diff
@@ -214,6 +214,12 @@ Fapi_CreateNv_Async(
                            policyPath);
     goto_if_error(r, "Set key flags for NV object", error_cleanup);
 
+    if (nvCmd->public_templ.public.nvIndex) {
+        /* NV index was defined by the user, has to be checked. */
+        r = ifapi_check_nv_index(path, nvCmd->public_templ.public.nvIndex);
+        goto_if_error(r, "Check NV path and NV index", error_cleanup);
```

Comment:
This is checking just the index, not the path, right?

Reply:
It will be checked whether the index selected by the user is appropriate for the provided path. If no index is selected by the user an index appropriate for the path will be determined. I will describe this in the comment.